src/Npgsql/NpgsqlTypes/NpgsqlDbType.cs

```diff
@@ -532,4 +533,17 @@ internal BuiltInPostgresType(string name, uint oid)
             OID = oid;
         }
     }
+
+    /// <summary>
+    /// Defines extension methods for <see cref="NpgsqlDbType"/>.
+    /// </summary>
+    static class NpgsqlDbTypeExtensions
+    {
+        internal static string? GetPostgresTypeName(this NpgsqlDbType type)
+        {
+            return typeof(NpgsqlDbType)
+                .GetMember(type.ToString())[0]
```

Comment:
I don't think that's necessary since the supplied parameter is of the type of the enum, so it's safe to assume that the member will always be found.

Reply:
It won't work because `NpgsqlDbType` actually is a flagged enum. You should apply a mask first to remove `Array` and `Range` from the type. Plus a new method used only at a single place isn't a good idea.

@roji, is this code perf critical? If so I would suggest to have a lazy loaded dictionary.

Reply:
I see. I am adding new tests for Array and Range types and removing the mask. But extension types are also returning `null`. What is expected for these types?